src/orquestra/opt/optimizers/scipy_optimizer.py

```diff
@@ -14,6 +14,77 @@
 )
 from ..history.recorder import RecorderFactory
 from ..history.recorder import recorder as _recorder
+from .pso.continuous_pso_optimizer import _get_bounds_like_array
+
+
+class _CostFunctionWithBestValueType(type):
+    def __instancecheck__(cls, obj: object) -> bool:
+        if (
+            isinstance(obj, Callable)
+            and hasattr(obj, "best_value")
+            and hasattr(obj, "best_params")
+        ):
+            return True
+        return False
+
+
+class _CostFunctionWithBestValue(metaclass=_CostFunctionWithBestValueType):
+    best_value: float = np.inf
+    best_params: np.ndarray = np.empty(1)
+
+    def __init__(
+        self,
+        cost_function: Union[CallableWithGradient, Callable],
+        constraints: Sequence[Dict[str, Callable]],
```

Comment:
```suggestion
        constraints: Tuple[Dict[str, Callable]],
```
Just to keep typing consistent between optimizers. 😉

Reply:
If it is possible to have a `CostFunctionWithBestValue` with no constraints, it might be best to put the `Optional` type on there too.

Reply:
Done 😄